components/brave_sync/time_limited_words_unittest.cc

```diff
@@ -0,0 +1,185 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_sync/time_limited_words.h"
+
+#include <memory>
+
+#include "base/logging.h"
+#include "base/test/gtest_util.h"
+#include "base/time/time_override.h"
+#include "brave/components/brave_sync/qr_code_validator.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using base::subtle::ScopedTimeClockOverrides;
+
+namespace brave_sync {
+
+namespace {
+
+const char kValidSyncCode[] =
+    "fringe digital begin feed equal output proof cheap "
+    "exotic ill sure question trial squirrel glove celery "
+    "awkward push jelly logic broccoli almost grocery drift";
+
+const char kInvalidSyncCode[] =
+    "fringe digital begin feed equal output proof cheap "
+    "exotic ill sure question trial squirrel glove celery "
+    "awkward push jelly logic broccoli almost grocery driftZ";
+
+base::Time g_overridden_now;
+std::unique_ptr<ScopedTimeClockOverrides> OverrideWithTimeNow(
+    const base::Time& overridden_now) {
+  g_overridden_now = overridden_now;
+  return std::make_unique<ScopedTimeClockOverrides>(
+      []() { return g_overridden_now; }, nullptr, nullptr);
+}
+}  // namespace
+
+TEST(TimeLimitedWordsTest, GetRoundedDaysDiff) {
+  const base::Time time1 = QrCodeDataValidator::GetQRv1SunsetDay();
+
+  base::Time time2 = time1 + base::Hours(11);
+  EXPECT_EQ(TimeLimitedWords::GetRoundedDaysDiff(time1, time2), 0);
+
+  time2 = time1 + base::Hours(13);
+  EXPECT_EQ(TimeLimitedWords::GetRoundedDaysDiff(time1, time2), 1);
+  EXPECT_EQ(TimeLimitedWords::GetRoundedDaysDiff(time2, time1), -1);
+}
+
+TEST(TimeLimitedWordsTest, GetIndexByWord) {
+  EXPECT_EQ(TimeLimitedWords::GetIndexByWord("abandon"), 0);
+  EXPECT_EQ(TimeLimitedWords::GetIndexByWord("ability"), 1);
+  EXPECT_EQ(TimeLimitedWords::GetIndexByWord("not_bip39_word"), -1);
+}
+
+TEST(TimeLimitedWordsTest, GetWordByIndex) {
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(0), "abandon");
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(1), "ability");
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(2047), "zoo");
+  EXPECT_EQ(TimeLimitedWords::GetWordByIndex(2048), "abandon");
+}
+
+TEST(TimeLimitedWordsTest, GenerateForDate) {
+  EXPECT_EQ(std::string(kValidSyncCode) + " abandon",
```

Comment:
Here I would suggest using a different constant than the sunset value to make the test clearer. The first two test cases don't have anything to do with the sunset value.

Reply:
thanks, I fixed this at https://github.com/brave/brave-core/pull/12300/commits/4c59158b9070eacf74afa52c42c1930b85f942e4